tracker/event_tracker.go

```diff
@@ -67,28 +67,30 @@ func (e *EventTracker) Start() error {
 	}
 
 	go func() {
-		go func() {
-			if err := tt.Sync(context.Background()); err != nil {
-				e.logger.Error("Event tracker", "failed to sync", err)
-			}
-		}()
-
-		go func() {
-			for {
-				select {
-				case evnt := <-tt.EventCh:
-					if len(evnt.Removed) != 0 {
-						panic("this will not happen anymore after tracker v2")
-					}
-
-					for _, log := range evnt.Added {
-						e.subscriber.AddLog(log)
-					}
-				case <-tt.DoneCh:
-					e.logger.Info("Historical sync done")
+		if err := tt.Sync(ctx); err != nil {
+			e.logger.Error("failed to sync", "error", err)
+		}
+	}()
+
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+
+			case evnt := <-tt.EventCh:
+				if len(evnt.Removed) != 0 {
+					panic("this will not happen anymore after tracker v2")
```

Comment:
This panic should vanish, although we must agree how to resolve it properly, since not sure how to properly handle reorgs (e.g. is there a real chance that reorgs can happen on Ethereum post-merge).

However not sure what's the correct approach to change it, since it denotes that a reorg happened on a target chain, which can be either L1/Ethereum or L2/Polygon (depending which events we are tracking).
I'm aware that on Polygon chain, there shouldn't ever be reorgs and forks because of instant finality.
On the other hand, post-merge Ethereum shouldn't [face reorgs as before](https://hackmd.io/@vbuterin/reorg_wtf_slides#/), if any (?) (although it seems there is at least theoretical possibility to have it, since it still runs "weighted" [fork choice](https://ethereum.org/en/developers/docs/consensus-mechanisms/#:~:text=however%2C%20now%20that%20Ethereum%20runs%20on%20proof%2Dof%2Dstake%20it%20adopted%20an%20updated%20fork%2Dchoice%20algorithm%20that%20measures%20the%20%27weight%27%20of%20the%20chain.%20The%20weight%20is%20the%20accumulated%20sum%20of%20validator%20votes%2C%20weighted%20by%20validator%20staked%2Dether%20balances.) algorithm).
So the question boils down to should we worry about reorgs on the Ethereum Beacon chain?
Based on it, we have several options to consider what to do with that panic in event_tracker:
- remove entirely handling of Removed logs together with panic (https://github.com/0xPolygon/polygon-edge/blob/feature/v3-parity/tracker/event_tracker.go#L80)
- just add logging, but ignore the existence of Removed logs (consider only Added ones)
- preserve mentioned if and panic until further notice
- change logic how we are treating logs (such that we remove logs from polybft internal state, because it basically denotes that a reorg happened and given log is no longer a valid). If there can be reorgs, this is probably the best approach.